File: LeetCode/1295. Find Numbers with Even Number of Digits.py
# 1295. Find Numbers with Even Number of Digits

from typing import List


class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        output = 0
        for num in nums:
            num_str = str(num)
            if len(num_str) % 2 == 0:
                output += 1

        return output


# # test
# sol = Solution()
# print(sol.findNumbers([555,901,482,1771]))

File: LeetCode/1295. Find Numbers with Even Number of Digits.cpp
/*
1295. Find Numbers with Even Number of Digits

给一组整数，返回其中位数是偶数的数有多少个。
*/

class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int result = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (to_string(nums[i]).size() % 2 == 0)
                result++;
        }
        return result;
    }
};